flux-core fails to build for ppc64le in the buildfarm

```
Date: Tue, 13 Dec 2016 23:43:33 +0000
From: "Foraker, Jim" <foraker1@llnl.gov>
To: Jim Garlick <garlick@llnl.gov>
CC: "Mark A. Grondona" <mgrondona@llnl.gov>
Subject: flux-core fails to build for ppc64le in the buildfarm

     Sadly true.  Here’s a build log from trying to rebuild the current
ch6-chaotic version, 0.4.1-1.ch6:

http://tossbuild.llnl.gov/kojifiles/work/tasks/8401/28401/build.log

It looks like maybe architecture detection is going a little haywire.
Obviously not critical, but you’ll need to either fix it up or sprinkle
some ExclusiveArch magic into your spec file the next time you build on
ch6-chaotic.

     Jim
```

build log with tar output redacted
```
Mock Version: 1.2.21
Mock Version: 1.2.21
ENTER ['do'](['bash', '--login', '-c', '/usr/bin/rpmbuild -bs --target ppc64le --nodeps /builddir/build/SPECS/flux-core.spec'], chrootPath='/var/lib/mock/ch6-chaotic-build-7695-5593/root'shell=FalseprintOutput=Falseenv={'LANG': 'en_US.UTF-8', 'TERM': 'vt100', 'SHELL': '/bin/bash', 'HOSTNAME': 'mock', 'PROMPT_COMMAND': 'printf "\\033]0;<mock-chroot>\\007"', 'HOME': '/builddir', 'PATH': '/usr/bin:/bin:/usr/sbin:/sbin', 'PS1': '<mock-chroot> \\s-\\v\\$ '}gid=135user='mockbuild'timeout=86400logger=<mockbuild.trace_decorator.getLog object at 0x10021ea1550>uid=995)
Executing command: ['bash', '--login', '-c', '/usr/bin/rpmbuild -bs --target ppc64le --nodeps /builddir/build/SPECS/flux-core.spec'] with env {'LANG': 'en_US.UTF-8', 'TERM': 'vt100', 'SHELL': '/bin/bash', 'HOSTNAME': 'mock', 'PROMPT_COMMAND': 'printf "\\033]0;<mock-chroot>\\007"', 'HOME': '/builddir', 'PATH': '/usr/bin:/bin:/usr/sbin:/sbin', 'PS1': '<mock-chroot> \\s-\\v\\$ '} and shell False
Building target platforms: ppc64le
Building for target ppc64le
Wrote: /builddir/build/SRPMS/flux-core-0.4.1-1.ch6.src.rpm
Child return code was: 0
ENTER ['do'](['bash', '--login', '-c', u'/usr/bin/rpmbuild -bb --target ppc64le --nodeps /builddir/build/SPECS/flux-core.spec'], chrootPath='/var/lib/mock/ch6-chaotic-build-7695-5593/root'shell=Falseuid=995env={'LANG': 'en_US.UTF-8', 'TERM': 'vt100', 'SHELL': '/bin/bash', 'HOSTNAME': 'mock', 'PROMPT_COMMAND': 'printf "\\033]0;<mock-chroot>\\007"', 'HOME': '/builddir', 'PATH': '/usr/bin:/bin:/usr/sbin:/sbin', 'PS1': '<mock-chroot> \\s-\\v\\$ '}gid=135user='mockbuild'timeout=86400private_network=Truelogger=<mockbuild.trace_decorator.getLog object at 0x10021ea1550>printOutput=False)
Executing command: ['bash', '--login', '-c', u'/usr/bin/rpmbuild -bb --target ppc64le --nodeps /builddir/build/SPECS/flux-core.spec'] with env {'LANG': 'en_US.UTF-8', 'TERM': 'vt100', 'SHELL': '/bin/bash', 'HOSTNAME': 'mock', 'PROMPT_COMMAND': 'printf "\\033]0;<mock-chroot>\\007"', 'HOME': '/builddir', 'PATH': '/usr/bin:/bin:/usr/sbin:/sbin', 'PS1': '<mock-chroot> \\s-\\v\\$ '} and shell False
Building target platforms: ppc64le
Building for target ppc64le
Executing(%prep): /bin/sh -e /var/tmp/rpm-tmp.1AskwD
+ umask 022
+ cd /builddir/build/BUILD
+ cd /builddir/build/BUILD
+ rm -rf flux-core-0.4.1
+ /usr/bin/gzip -dc /builddir/build/SOURCES/flux-core-0.4.1.tar.gz
+ /usr/bin/tar -xvvf -
drwxrwxr-x 1000/1000         0 2016-08-12 10:58 flux-core-0.4.1/
[snip]
-rwxrwxr-x 1000/1000      2082 2016-08-12 10:56 flux-core-0.4.1/etc/gen-cmdhelp.pl
+ STATUS=0
+ '[' 0 -ne 0 ']'
+ cd flux-core-0.4.1
+ /usr/bin/chmod -Rf a+rX,u+w,g-w,o-w .
+ exit 0
+ umask 022
+ cd /builddir/build/BUILD
+ cd flux-core-0.4.1
+ KOJI_CFLAGS='-O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches  -m64 -mtune=generic'
+ CFLAGS='-O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches  -m64 -mtune=generic -Wno-error=strict-aliasing'
+ CFLAGS='-O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches  -m64 -mtune=generic -Wno-error=strict-aliasing'
+ export CFLAGS
+ CXXFLAGS='-O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches   -m64 -mcpu=power8 -mtune=power8'
+ export CXXFLAGS
+ FFLAGS='-O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches   -m64 -mcpu=power8 -mtune=power8 -I/usr/lib64/gfortran/modules'
+ export FFLAGS
+ FCFLAGS='-O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches   -m64 -mcpu=power8 -mtune=power8 -I/usr/lib64/gfortran/modules'
+ export FCFLAGS
+ LDFLAGS='-Wl,-z,relro '
+ export LDFLAGS
+ '[' 1 == 1 ']'
+ '[' ppc64le == ppc64le ']'
+ /usr/lib/rpm/redhat/libtool-handle-ppc64le.sh
Executing(%build): /bin/sh -e /var/tmp/rpm-tmp.fBft74
Broken -m elf64ppc use in ./configure should handle elf64lppc.
Attempting automatic fix.
Fixed ./configure for ld -m ppc64le support.
++ find . -name config.guess -o -name config.sub
+ for i in '$(find . -name config.guess -o -name config.sub)'
++ basename ./config/config.sub
+ '[' -f /usr/lib/rpm/redhat/config.sub ']'
+ /usr/bin/rm -f ./config/config.sub
++ basename ./config/config.sub
+ /usr/bin/cp -fv /usr/lib/rpm/redhat/config.sub ./config/config.sub
'/usr/lib/rpm/redhat/config.sub' -> './config/config.sub'
+ for i in '$(find . -name config.guess -o -name config.sub)'
++ basename ./config/config.guess
+ '[' -f /usr/lib/rpm/redhat/config.guess ']'
+ /usr/bin/rm -f ./config/config.guess
++ basename ./config/config.guess
+ /usr/bin/cp -fv /usr/lib/rpm/redhat/config.guess ./config/config.guess
'/usr/lib/rpm/redhat/config.guess' -> './config/config.guess'
+ ./configure --build=ppc64le-koji-linux-gnu --host=ppc64le-koji-linux-gnu --program-prefix= --disable-dependency-tracking --prefix=/usr --exec-prefix=/usr --bindir=/usr/bin --sbindir=/usr/sbin --sysconfdir=/etc --datadir=/usr/share --includedir=/usr/include --libdir=/usr/lib64 --libexecdir=/usr/libexec --localstatedir=/var --sharedstatedir=/var/lib --mandir=/usr/share/man --infodir=/usr/share/info --disable-static
checking build system type... powerpc64le-koji-linux-gnu
checking host system type... powerpc64le-koji-linux-gnu
checking target system type... powerpc64le-koji-linux-gnu
checking installation directory variables... yes
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /usr/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether to enable maintainer-specific portions of Makefiles... no
checking for ppc64le-koji-linux-gnu-pkg-config... no
checking for pkg-config... /usr/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for style of include used by make... GNU
checking for ppc64le-koji-linux-gnu-gcc... no
checking for gcc... gcc
checking whether the C compiler works... no
configure: error: in `/builddir/build/BUILD/flux-core-0.4.1':
configure: error: C compiler cannot create executables
See `config.log' for more details
RPM build errors:
error: Bad exit status from /var/tmp/rpm-tmp.fBft74 (%build)
    Bad exit status from /var/tmp/rpm-tmp.fBft74 (%build)
Child return code was: 1
EXCEPTION: [Error()]
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/mockbuild/trace_decorator.py", line 88, in trace
    result = func(*args, **kw)
  File "/usr/lib/python2.7/site-packages/mockbuild/util.py", line 562, in do
    raise exception.Error("Command failed. See logs for output.\n # %s" % (command,), child.returncode)
Error: Command failed. See logs for output.
 # bash --login -c /usr/bin/rpmbuild -bb --target ppc64le --nodeps /builddir/build/SPECS/flux-core.spec
```

--- a/src/test/travis-dep-builder.sh
+++ b/src/test/travis-dep-builder.sh
@@ -16,7 +16,10 @@ https://github.com/zeromq/czmq/archive/v3.0.2.tar.gz \
 http://downloads.sourceforge.net/ltp/lcov-1.10.tar.gz \
 http://www.open-mpi.org/software/hwloc/v1.11/downloads/hwloc-1.11.0.tar.gz \
 http://www.digip.org/jansson/releases/jansson-2.9.tar.gz \
-http://www.mpich.org/static/downloads/3.1.4/mpich-3.1.4.tar.gz"
+http://www.mpich.org/static/downloads/3.1.4/mpich-3.1.4.tar.gz \
+https://ftp.gnu.org/gnu/autoconf/autoconf-2.69.tar.gz \
+https://ftp.gnu.org/gnu/automake/automake-1.15.tar.gz \
+https://ftp.gnu.org/gnu/libtool/libtool-2.4.6.tar.gz"
 
 declare -A extra_configure_opts=(\
 ["mpich-3.1.4"]="--disable-fortran --disable-cxx --disable-maintainer-mode --disable-dependency-tracking --enable-shared --disable-wrapper-rpath" \

--- a/src/test/travis-dep-builder.sh
+++ b/src/test/travis-dep-builder.sh
@@ -102,7 +102,7 @@ print_env () {
     echo "export CPPFLAGS=-I${prefix}/include"
     echo "export LDFLAGS=-L${prefix}/lib"
     echo "export PKG_CONFIG_PATH=${prefix}/lib/pkgconfig:${prefix}/share/pkgconfig"
-    echo "export PATH=${PATH}:${HOME}/.local/bin:${HOME}/local/usr/bin:${HOME}/local/bin"
+    echo "export PATH=${HOME}/.local/bin:${HOME}/local/usr/bin:${prefix}/bin:${PATH}"
     luarocks path --bin
 }
 

